aioshelly/wsrpc.py

```diff
@@ -206,34 +258,62 @@ def connected(self) -> bool:
 
     async def call(
         self, method: str, params: dict[str, Any] | None = None, timeout: int = 10
+    ) -> dict[str, Any]:
+        """Websocket RPC call."""
+        # Try request with initial/last call auth data
+        resp = await self._rpc_call(method, params, timeout)
+        if "result" in resp:
+            return cast(dict, resp["result"])
+
+        try:
+            code, msg = resp["error"]["code"], resp["error"]["message"]
+        except KeyError as err:
+            raise RPCError(f"bad response: {resp}") from err
+
+        if code != 401:
+            raise JSONRPCError(code, msg)
+        if self._auth_data is None:
+            raise InvalidAuthError(code, msg)
+
+        # Update auth from response and try with new auth data
+        auth = json.loads(msg)
+        self._session.auth = self._auth_data.get_auth(auth["nonce"], auth.get("nc", 1))
+        resp = await self._rpc_call(method, params, timeout)
```

Comment:
Better to call yourself again but now set a keyword arg to not refresh auth. Easier to keep the code in sync.

```suggestion
        return await self.call(method, params, timeout, handle_auth=False)
```